### PR TITLE
[skip ci]Archive log ignore exception

### PR DESCRIPTION
**Pull request by PengTian0**
The log won't be archived if exception raised before the archive step.
This PR will archive log ignore exception.

### CODE REVIEW

**Comment by PengTian0** [comment]
@changev @yyscamper @anhou

File: jobs/function_test/function_test.groovy [hunk 1]
```diff
@@ -69,16 +69,24 @@ def function_test(String test_name, String label_name, String TEST_GROUP, Boolea
...
                         
                         def artifact_dir = test_name.replaceAll(' ', '-')
-                        sh '''#!/bin/bash -ex
+                        sh '''#!/bin/bash -x
                         mkdir '''+"$artifact_dir"+'''
-                        cp RackHD/test/*.xml '''+"$artifact_dir" +'''
-                        ./build-config/jobs/function_test/cleanup.sh
                         ./build-config/post-deploy.sh
-                        cp build-deps/*.log '''+"$artifact_dir"+'''
+                        find build-deps/ -maxdepth 1 -name "*.log" > files.txt
```
Comment (changev) [inline jobs/function_test/function_test.groovy]: Minor question. Is this different with ```ls *.log```?

File: jobs/function_test/function_test.groovy [hunk 2]
```diff
@@ -69,16 +69,24 @@ def function_test(String test_name, String label_name, String TEST_GROUP, Boolea
...
+                        files=$( paste -s -d ' ' files.txt )
```
Comment (changev) [inline jobs/function_test/function_test.groovy]: If ```files=`find build-deps/ -maxdepth 1 -name "*.log"` ``` can satisfy the reqirments?

File: jobs/function_test/function_test.groovy (hunk 1 above)
Comment (PengTian0) [reply jobs/function_test/function_test.groovy]: Thanks!
Fixed.

File: jobs/function_test/function_test.groovy (hunk 2 above)
Comment (PengTian0) [reply jobs/function_test/function_test.groovy]: Thanks!
Fixed.

**Comment by changev** [comment]
:+1:  :100:

**Review by yyscamper — APPROVED** [review]

**Pull request merged**